scripts/base/frameworks/cluster/nodes/manager.zeek

```diff
@@ -23,3 +23,53 @@ redef Log::default_rotation_interval = 24 hrs;
 ## Use the cluster's delete-log script.
 redef Log::default_rotation_postprocessor_cmd = "delete-log";
 @endif
+
+module Cluster;
+
+global fully_connected_nodes_pending: set[string];
+
+event zeek_init()
+	{
+	fully_connected_nodes_pending = table_keys(nodes);
+	}
+
+event Cluster::node_fully_connected(name: string, id: string, redispatched: bool)
+	{
+	# If a node redispatched the event, the cluster has already been seen connected.
+	seen_cluster_connected = redispatched ? T : seen_cluster_connected;
+
+	delete fully_connected_nodes_pending[name];
+	if ( !seen_cluster_connected && |fully_connected_nodes_pending| == 0 )
+		{
+		event Cluster::cluster_connected();
+
+		for ( topic in broadcast_topics )
+			Broker::publish(topic, Cluster::cluster_connected);
+		}
```

Comment:
Is this missing the targeted `Cluster::cluster_connected` publish to the node_topic because it's not yet clear whether/how to do this?

E.g, something like this for an individual node that comes after cluster_connected happened?
```
if ( seen_cluster_connected && !redispatched)
   {
   Broker::publish(Cluster::node_topic(name), Cluster::cluster_connected)
   }
```
And separately, a manager raising `Cluster::cluster_connected()` for itself once if it finds someone else saw `cluster_connected` before. Think the `||` suggestion above could be changed to an `if` where this happens then.


Separately wondering a bit if an explicit `did_publish_cluster_connected` global known to the manager here would make things easier to read.

Reply:
The current state just implements variant 2, because I had a slight preference for that approach ;)

> Separately wondering a bit if an explicit did_publish_cluster_connected global known to the manager here would make things easier to read.

I think if we go with `cluster_started`, `seen_cluster_connected` becomes `is_cluster_started` and things should be readable without introducing more variables.